### DISCUSSION TITLE
How to replace map to unordered_map

### DISCUSSION THREAD

**Discussion by Eagleyangs** [open-source-parsers/jsoncpp#1410] [Q&A]
When I replace map with unordered_map, a compilation error occurs indicating that illegal expressions cannot be used.

Modified:
      //typedef std::map ObjectValues;
      typedef std::unordered_map ObjectValues;

      //return (*value_.map_) < (*other.value_.map_);
      return (*value_.map_) ! = (*other.value_.map_);

![20220608164537](https://user-images.githubusercontent.com/25238164/172573585-b86b3a37-0f04-46d6-a030-783eb6284fb3.png)

How can I do to make sure it compiles correctly and functions correctly?

**Comment by BillyDonahue** [comment]
This needs to be more precisely phrased to get an answer. It sounds like you're trying to mess with the internals of `Json::Value`.
I wouldn't recommend doing that unless you have a lot of experience with that kind of code.

**Reply by PatrickYuSuper** [reply]
Can JsonCPP support un-ordered map? The usage scenario is to except JSON object and write to a JSON file and preserving its original content/order.

**Reply by BillyDonahue** [reply]
Even if Jsoncpp did support unordered_map as the storage for object values, unordered_map doesn't preserve insertion order.
